osm2gtfs/core/osm_connector.py

```diff
@@ -345,57 +367,171 @@ def _build_route_variant(self, route_variant, query_result_set, rm=None):
                     otype = "way"
 
                 else:
-                    raise RuntimeError("Unknown type: " + str(stop_candidate))
+                    sys.stderr.write("Unknown type of itinerary member: " +
+                                     str(stop_candidate) + "\n")
 
                 stops.append(otype + "/" + str(stop_candidate.ref))
 
+        if route_master:
+            parent_identifier = osm_type + "/" + str(route_master.id)
+        else:
+            parent_identifier = None
+
+        # Move to Elements class, once attributes with defaults play well with
+        # inheritance https://github.com/python-attrs/attrs/issues/38
+        osm_url = "https://osm.org/" + str(
+            osm_type) + "/" + str(route_variant.id)
+        if 'name' in route_variant.tags:
+            name = route_variant.tags['name']
+        elif 'ref' in route_variant.tags:
+            name = route_variant.tags['ref']
+        else:
+            name = None
+
         shape = self._generate_shape(route_variant, query_result_set)
-        rv = Route(route_variant.id, fr, to, stops,
-                   rm, ref, name, shape, travel_time)
-        print(rv)
+
+        rv = Itinerary(osm_id=route_variant.id, osm_type=osm_type,
+                       osm_url=osm_url, name=name, tags=route_variant.tags,
+                       route_id=ref, shape=shape, line=parent_identifier,
```

Comment:
Isn't better to rename the `Itinerary.route_id` as `Itinerary.ref`?

Reply:
No, `route_id` is the terminology of GTFS, which is more dominant in the naming of class variables and attributes.

Reply:
ok.